santoku/sql/mysql_handler.py

```diff
@@ -0,0 +1,156 @@
+from typing import List, Tuple, Dict, Any
+
+import pandas as pd
+import mysql.connector
+
+from santoku.aws.secrets_manager_handler import SecretsManagerHandler
+
+
+class MySQLHandler:
+    """
+    Manage MySQL interactions, the simplest of which is to query a particular table.
+    This makes use of the official MySQL connector. More information at
+    https://dev.mysql.com/doc/connector-python/en/
+
+    """
+
+    def __init__(self, user: str, password: str, host: str, database: str) -> None:
+        """
+        Base constructor. Use this to connect to the MySQL server.
+        For alternative methods of authentication use the given class methods.
+
+        Parameters
+        ----------
+        user : str
+            The MySQL user name.
+        password : str
+            The password of the MySQL user.
+        host : str
+            Hostname or location of the MySQL server.
+        database : str
+            The database in the MySQL server to be connected to.
+
+        Raises
+        ------
+        mysql.connector.errors.ProgrammingError
+            Raised if credentials are not correctly specified or the database does not exsit.
+
+        mysql.connector.errors.InterfaceError
+            Raised if the connection to the MySQL server host cannot be done.
+
+        Notes
+        -----
+        More information on MySQL connection: [1].
+
+        References
+        ----------
+        [1] :
+        https://dev.mysql.com/doc/connector-python/en/connector-python-example-connecting.html
+
+        """
+        self.mysql_cnx = mysql.connector.connect(
```

Comment:
the class name already specifies this is MySQL we are using, I'd simply call this `cnx` or, more descriptibly, `db_connector`